src/nlp.jl

```diff
@@ -178,7 +178,7 @@ function _nlp_objective_function(model::GenericModel)
     if model.nlp_model === nothing
         return nothing
     end
-    return model.nlp_model.objective
+    return something(model.nlp_model).objective
```

Comment:
It's surprising the compiler isn't able to figure this one out

Reply:
Part of this is because `model.nlp_model` calls `getproperty`, which may do arbitrary things. 

This is another cause where the specific concrete call of a particular `GenericModel{T}` might be able to, but we can't figure it out statically based on `(::GenericModel)`.